src/NSwag.Commands/Commands/SwaggerGeneration/SwaggerGeneratorCommandBase.cs

```diff
@@ -203,25 +208,54 @@ public async Task<TSettings> CreateSettingsAsync(AssemblyLoader.AssemblyLoader a
             }
         }
 
-        protected TestServer CreateTestServer(Type startupType)
+        protected async Task<TestServer> CreateTestServerAsync(AssemblyLoader.AssemblyLoader assemblyLoader)
         {
-            return new TestServer(new WebHostBuilder().UseStartup(startupType));
-        }
-
-        protected async Task<Type> GetStartupTypeAsync(AssemblyLoader.AssemblyLoader assemblyLoader)
-        {
-            Type startupType;
-
-            if (string.IsNullOrEmpty(StartupType))
+            if (!string.IsNullOrEmpty(CreateWebHostBuilderMethod))
             {
-                // TODO: Implement proper auto discovery
-                var assemblies = await LoadAssembliesAsync(AssemblyPaths, assemblyLoader).ConfigureAwait(false);
-                startupType = assemblies.First().ExportedTypes.First(t => t.Name == "Startup");
+                // Load configured CreateWebHostBuilder method from program type
+                var segments = CreateWebHostBuilderMethod.Split('.');
+                var programTypeName = string.Join(".", segments.Take(segments.Length - 1));
+                var programType = assemblyLoader.GetType(programTypeName);
```

Comment:
Do we want to trow a meaningful exception when the type is not found?

Reply:
Added support for method without params + improved exception handling